### PR TITLE
clang6 fix in DNS.cc

### PR DESCRIPTION
**Pull request by a-canary**
dont copy to same address

### CODE REVIEW

**Comment by bryancall** [comment]
[approve ci clang-analyzer]

**Comment by zwoop** [comment]
[approve ci]

**Comment by a-canary** [comment]
[approve ci autest]

File: iocore/dns/DNS.cc [hunk 1]
```diff
@@ -474,7 +474,7 @@ DNSHandler::open_con(sockaddr const *target, bool failed, int icon, bool over_tc
   ip_port_text_buffer ip_text;
   PollDescriptor *pd = get_PollDescriptor(dnsProcessor.thread);
 
-  if (!icon && target) {
+  if (!icon && target && &ip.sa != target) {
```
Comment (zwoop) [inline iocore/dns/DNS.cc]: Wouldn't it be better to move this test to ats_ip_copy() ?

Comment (SolidWallOfCode) [reply iocore/dns/DNS.cc]: Yes.

Comment (a-canary) [reply iocore/dns/DNS.cc]: you're right, i don't like c. I shouldn't mimic it.

**Pull request merged**